spec/services/gift_sticker_service_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe GiftStickersService do
+  describe '.call' do
+    context 'Users in the incompleted state' do
+      let(:user_with_early_receipt_2prs) do
+        FactoryBot.create(:user, :incompleted)
+      end
+      let(:user_with_early_receipt_3prs) do
+        FactoryBot.create(:user, :incompleted)
+      end
+      let(:user_with_late_receipt_2prs) do
+        FactoryBot.create(:user, :incompleted)
+      end
+      let(:user_with_late_receipt_3prs) do
+        FactoryBot.create(:user, :incompleted)
+      end
+
+      before do
+        user_with_early_receipt_2prs
+          .update(receipt: UserReceiptHelper.receipt[:early_receipt_2_prs])
+        user_with_early_receipt_3prs
+          .update(receipt: UserReceiptHelper.receipt[:early_receipt_3_prs])
+        user_with_late_receipt_2prs
+          .update(receipt: UserReceiptHelper.receipt[:late_receipt_2_prs])
+        user_with_late_receipt_3prs
+          .update(receipt: UserReceiptHelper.receipt[:late_receipt_3_prs])
```

Comment:
This starter data should be updated to reflect the logic that a user with the earliest date on their latest PR should be ranked first for users with that number of PRs.

Reply:
This is actually already the case. My main mistake was just grabbing the first created PR per user as opposed to the last in the `GiftService` . However, the user with the earliest date on their latest PR will always be ranked first. So if there is only one coupon `user_with_early_receipt_3prs` (meaning user with score of 3, who's last PR has earliest PR created_at time) will be the one getting the coupon.